### PR TITLE
Remove uses of deprecated ioutil package

### LINKED ISSUE #605: Update deprecated io/ioutil package
As of go1.16 the io/ioutil package has been deprecated and needs to replaced by either `os` or `io` package.

The lint workflow has been failing due to this issue

### PR DESCRIPTION
**Pull request by ernestask**
See individual commits.

Closes #605

### CODE REVIEW

File: checks/folder.go [hunk 1]
```diff
@@ -106,12 +105,17 @@ func getLocalFolderCheck(path string, filter v1.FolderFilter) (*FolderCheck, err
...
-		if file.IsDir() || !_filter.Filter(file) {
+	for _, entry := range entries {
+		info, err := entry.Info()
+		if err != nil {
+			return nil, err
+		}
+
+		if !_filter.Filter(info) {
 			continue
 		}
```
Comment (ernestask) [inline checks/folder.go:116]: I dropped the `IsDir()` check, because it is already done in the filter.

**Comment by ernestask** [comment]
I’m guessing the issue with `github-test-expression-pass` fails due to the test being run on a fork?

File: checks/folder.go [hunk 2]
```diff
@@ -106,12 +105,17 @@ func getLocalFolderCheck(path string, filter v1.FolderFilter) (*FolderCheck, err
...
 
-	for _, file := range files {
-		if file.IsDir() || !_filter.Filter(file) {
+	for _, entry := range entries {
+		info, err := entry.Info()
+		if err != nil {
+			return nil, err
+		}
+
+		if !_filter.Filter(info) {
```
Comment (moshloop) [inline checks/folder.go:114]:
```suggestion
		if file.IsDir || !_filter.Filter(info) {
```

File: checks/folder.go [hunk 3]
```diff
@@ -106,12 +105,17 @@ func getLocalFolderCheck(path string, filter v1.FolderFilter) (*FolderCheck, err
 		return &FolderCheck{Oldest: info, Newest: info}, nil
 	}
 
-	for _, file := range files {
-		if file.IsDir() || !_filter.Filter(file) {
```
[the remark this answers is not in the document]
Comment (ernestask) [reply checks/folder.go:110]: https://github.com/flanksource/canary-checker/pull/612#discussion_r972100844

File: checks/folder.go (hunk 2 above)
Comment (ernestask) [reply checks/folder.go]: This adds back the duplicate check.

**Pull request closed without merging**